payments-core/src/main/java/com/stripe/android/model/StripeIntent.kt

```diff
@@ -172,6 +172,9 @@ sealed interface StripeIntent : StripeModel {
     }
 
     sealed class NextActionData : StripeModel {
+        interface DisplayVoucherDetails {
```

Comment:
You could also add another `val paymentMethodType: PaymentMethod.Type` here, and manually add it to each of the deserialized types. That would allow you to continue to have the data in the error reporter.

Reply:
It's a little weird though for `DisplayOxxoDetails` to need to have a payment method property to know that it's related to Oxxo though, right? It's right there in the name. Maybe there's something I should be doing with reflection here

Reply:
It's a little weird, I'd think about the type (DisplayOxxoDetails) telling the interface (DisplayVoucherDetails) some metadata about the implementation.

Reply:
Took a stab at using the `NextActionType` on the `authenticatable` instead. Lmk what you think!